azure/util.py

```diff
@@ -0,0 +1,309 @@
+        print("Running {}".format(cmd))
+        process = subprocess.Popen(
+            cmd,
+            universal_newlines=True,
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            env=os.environ,
+        )
+        for line in process.stdout:
+            print(line)
+            output.append(line)
+        for err in process.stderr:
+            print(err)
+    return output
+
+
+# Configures job yaml file based on user inputs
+def configure_yaml(args):
+    SAMPLE_YAML_FILE = os.path.join(PETCTL_DIR, "config/sample_specs.yaml")
+    result_yaml_file = os.path.join(PETCTL_DIR, "config/", "azure-pytorch-elastic.yaml")
+
+    print("Configuring job yaml ", result_yaml_file)
+
+    with open(SAMPLE_YAML_FILE) as f:
+        data = yaml.load(f)
+
+    data["spec"]["parallelism"] = args.max_size
+    data["spec"]["template"]["spec"]["containers"][0]["env"].extend(
+        [
+            dict([("name", "JOB_ID"), ("value", str(uuid.uuid1()) + "_" + args.name)]),
+            dict([("name", "MIN_SIZE"), ("value", str(args.min_size))]),
+            dict([("name", "MAX_SIZE"), ("value", str(args.max_size))]),
+        ]
+    )
+
+    yaml.dump(data, open(result_yaml_file, "w"))
+
+# Configures job yaml file based on user docker image
+def configure_yaml_storage(container_name):
+    yaml_file = os.path.join(PETCTL_DIR, "config/azure-pytorch-elastic.yaml")
+
+    print("Configuring job yaml ", yaml_file)
+
+    with open(yaml_file) as f:
+        data = yaml.load(f)
+
+    data["spec"]["template"]["spec"]["volumes"][0]["flexVolume"]["options"]["container"] = container_name
+
+    yaml.dump(data, open(yaml_file, "w"))
+
+# Configures job yaml file based on user docker image
+def configure_yaml_docker(image_name):
+    yaml_file = os.path.join(PETCTL_DIR, "config/azure-pytorch-elastic.yaml")
+
+    print("Configuring job yaml ", yaml_file)
+
+    with open(yaml_file) as f:
+        data = yaml.load(f)
+
+    data["spec"]["template"]["spec"]["containers"][0]["image"] = image_name
+
+    yaml.dump(data, open(yaml_file, "w"))
+
+# Configures kubernetes json file based on user inputs
+def configure_json(args):
+    KUBERNETES_JSON_FILE = os.path.join(PETCTL_DIR, "config/kubernetes.json")
+    result_json_file = os.path.join(PETCTL_DIR, "config/", "kubernetes.json")
+
+    print("Configuring kubernetes specs ", result_json_file)
+
+    with open(KUBERNETES_JSON_FILE) as f:
+        data = json.load(f)
+    data["properties"]["masterProfile"]["count"] = 1
+    data["properties"]["agentPoolProfiles"][0]["count"] = args.min_size
+    data["properties"]["masterProfile"]["vmSize"] = args.master_vm
+    data["properties"]["agentPoolProfiles"][0]["vmSize"] = args.worker_vm
+
+    json.dump(data, open(result_json_file, "w"), indent=4)
+
+
+# Log in to Azure
+def azure_login():
+    check_cmd = "az account show"
+    p = subprocess.Popen(check_cmd, shell=True, stdout=subprocess.PIPE, env=os.environ)
```

Comment:
I think you have a wrapper around subprocess already - perhaps use that?

Reply:
I removed the azure_login method.